Makefile

```diff
@@ -17,6 +17,12 @@ DOCS_CACHE_DIR = ${PWD}/docs/openapi
 TMP_DIR = $(shell mktemp -d "$${TMPDIR:-/tmp}/tmp.XXXXX")
 CURRENT_UID := $(shell id -u)
 CURRENT_GID := $(shell id -g)
+# Determine if Docker is being used via Docker Desktop, if so we need to tell testcontainers to
+# use the Docker Desktop host. We can't just always set this because GitHub Actions will fail.
+DOCKER_PLATFORM := $(shell docker version -f json | jq .Server.Platform.Name)
```

Comment:
Non-blocking, but I am not a big fan of this. It assumes that the only people running this locally are using Docker Desktop, which excludes Linux users and other container file technologies like Podman.

We can merge it now to unblock, but let's keep it in mind that we should find a better solution

Reply:
What is the improvement you're looking for?

This is required because of a Testcontainers requirement with Docker Desktop. They don't call out this requirement for any other container technology and I don't really have the means to test. I'm not fully understanding the issue here as we already use the docker CLI to build the SDKs and this change shouldn't introduce any requirement for Docker Desktop, just make things work for those who use it.

There could be an issue with the Docker socket we mount for Testcontainers but I feel like that's something that should have been ironed out prior to accepting the change to use Testcontainers and I think we could potentially do something similar here where we sniff the type of backend being used and change dependent on that?